src/components/forms/commodity/CommodityForm.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { classValidatorResolver } from '@hookform/resolvers/class-validator';
+import { SingleValue } from 'react-select';
+
+import { Commodity } from '@/book/entities';
+import Selector from '@/components/selectors/Selector';
+
+const resolver = classValidatorResolver(Commodity, { validator: { stopAtFirstError: true } });
+
+export type FormValues = {
+  namespace: string,
+  mnemonic: string,
+};
+
+export type CommodityFormProps = {
+  onSave: Function,
+};
+
+const BASE_CURRENCIES = [
+  { label: 'EUR' },
+  { label: 'USD' },
+  { label: 'SGD' },
+  { label: 'GBP' },
+];
```

Comment:
We probably should expand this as when it's single currency, it's just a name. Things get more complex when more currencies are added.